spyder/plugins/ipythonconsole.py

```diff
@@ -944,7 +946,8 @@ def create_new_client(self, give_focus=True, filename=''):
                               additional_options=self.additional_options(),
                               interpreter_versions=self.interpreter_versions(),
                               connection_file=cf,
-                              menu_actions=self.menu_actions)
+                              menu_actions=self.menu_actions,
+                              stderr_dir=stderr_dir)

```

Comment:
Instead of doing this, please change it to

    if self.testing:
        client.stderr_dir = self.test_dir